### PR TITLE
Fix for clustering

### PR DESCRIPTION
**Pull request by ogibson**
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-779

#### Link to staging deployment URL 

https://ui-oliver-w99p35.scp-staging.biomage.net/experiments/e52b39624588791a7889e39c617f669e/data-exploration

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ X] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [X ] Photo of a cute animal attached to this PR
image

### CODE REVIEW

File: r/src/embedding.r [hunk 1]
```diff
@@ -44,8 +44,30 @@ runEmbedding <- function(req) {
         df_embedding <- Embeddings(data, reduction = type)
     }
     # Order embedding by cells id in ascending form
-    df_embedding <- df_embedding[rownames(data@meta.data[order(data@meta.data$cells_id), ]), ]
```
Comment (marcellp) [inline r/src/embedding.r:47]: I assume this needs to be uncommented here.

**Comment by marcellp** [comment]
I would probably merge this change in with Juanlu's changes. They already have the tests disabled and I think he did some other stuff as well here.

File: r/src/embedding.r (hunk 1 above)
Comment (ogibson-biomage) [reply r/src/embedding.r:47]: No, it's ok that it's deleted.

Comment (alexvpickering) [reply r/src/embedding.r:47]: Nope it should be good, we remake it and store it in `tmp`

File: r/src/embedding.r [hunk 2]
```diff
@@ -44,8 +44,30 @@ runEmbedding <- function(req) {
...
+    tmp <- lapply(row.names(tmp), function(rname) {
+        umap1 <- tmp[rname, 'UMAP_1']
+        umap2 <- tmp[rname, 'UMAP_2']
+         if(is.na(umap1)) {
+            return(NA) 
+         } else {
+            return(c(umap1, umap2))
+         }
+    })
+    return(tmp)
```
Comment (alexvpickering) [inline r/src/embedding.r]: The following vectorized implementation is much more performant (the `lapply` so painfully slow) and cleaner:
I tested it locally in Rstudio with `e52...` and it addresses the same bug that you found in my original code.

```suggestion
    ids <- data@meta.data$cells_id+1
    idmax <- max(ids)
    ridx <- seq_len(idmax)
    res <- data.frame(row = ridx, dim1 = NA, dim2 = NA)
    res[ids, ]$dim1 <- df_embedding[,1]
    res[ids, ]$dim2 <- df_embedding[,2]
    res$row <- NULL
    # convert to list of vectors
    na.res <- is.na(res$dim1)
    res <- data.frame(t(res))
    res <- as.list(res)
    res[na.res] <- NA
    res <- unname(res)
    return(res)
```

File: r/src/embedding.r [hunk 3]
```diff
@@ -44,8 +44,30 @@ runEmbedding <- function(req) {
...
 
-    return(df_embedding)
+
+    # Order embedding by cells id in ascending form
+    #df_embedding <- df_embedding[rownames(data@meta.data[order(data@meta.data$cells_id), ]), ]
+
+    #return(df_embedding)
```
Comment (alexvpickering) [inline r/src/embedding.r]:
```suggestion
```

not needed

File: r/src/embedding.r (hunk 1 above)
Comment (ogibson) [reply r/src/embedding.r:47]: Tomorrow I'll add the code you sent me Alex

File: r/src/embedding.r (hunk 2 above)
Comment (ogibson) [reply r/src/embedding.r]: Great I'll fix this tomorrow!

File: r/src/embedding.r (hunk 1 above)
Comment (ogibson) [reply r/src/embedding.r:47]: I've actually uncommented this. The cell list order could be different than cellIds (I can't imagine how, but it's a possibility) so sorting the df should fix that.

**Review by marcellp — APPROVED** [review]
lgtm

**Pull request merged**